docs/book/how-to/steps-pipelines/advanced_features.md

```diff
@@ -303,6 +303,189 @@ The fan-in, fan-out method has the following limitations:
+
+```python
+from typing import List, Optional
+from uuid import UUID
+import time
+
+from zenml import step, pipeline, Client
+from zenml.client import Client
+
+
+@step
+def load_relevant_chunks() -> List[str]:
+    """Load chunk identifiers from database or other dynamic source."""
+    # Example: Query database for chunk IDs
+    # In practice, this could be a database query, API call, etc.
+    return ["chunk_1", "chunk_2", "chunk_3", "chunk_4"]
+
+
+@step
+def trigger_chunk_processing(
+    chunks: List[str], 
+    template_id: Optional[UUID] = None
+) -> List[UUID]:
+    """Trigger multiple pipeline runs for each chunk and wait for completion."""
+    client = Client()
+    
+    # Use template ID if provided, otherwise use pipeline name
+    pipeline_name = None if template_id else "chunk_processing_pipeline"
+    
+    # Trigger all chunk processing runs
+    run_ids = []
+    for chunk_id in chunks:
+        run_config = {
+            "steps": {
+                "process_chunk": {
+                    "parameters": {
+                        "chunk_id": chunk_id
+                    }
+                }
+            }
+        }
+        
+        run = client.trigger_pipeline(
+            template_id=template_id,
+            pipeline_name_or_id=pipeline_name,
+            run_configuration=run_config,
+            synchronous=False  # Run asynchronously
+        )
+        run_ids.append(run.id)
+    
+    # Wait for all runs to complete
+    print(f"Waiting for {len(run_ids)} chunk processing runs to complete...")
+    completed_runs = set()  # Cache completed runs to avoid re-fetching
+    while True:
+        # Only check runs that haven't completed yet
+        pending_runs = [run_id for run_id in run_ids if run_id not in completed_runs]
+        
+        for run_id in pending_runs:
+            run = client.get_pipeline_run(run_id)
+            if run.status.is_finished:
+                completed_runs.add(run_id)
+        
+        if len(completed_runs) == len(run_ids):
+            print("All chunk processing runs completed!")
+            break
+        
+        print(f"Completed: {len(completed_runs)}/{len(run_ids)} runs")
+        time.sleep(10)  # Wait 10 seconds before checking again
+    
+    return run_ids
+
+
+@step
+def aggregate_results(run_ids: List[UUID]) -> dict:
+    """Aggregate results from all chunk processing runs."""
+    client = Client()
+    aggregated_results = {}
+    failed_runs = []
+    
+    for run_id in run_ids:
+        run = client.get_pipeline_run(run_id)
+        
+        # Check if run succeeded
+        if run.status.is_failed:
```

Comment:
This `is_failed` doesn't exit I think? I think you should try to run this once at least to see if the code works